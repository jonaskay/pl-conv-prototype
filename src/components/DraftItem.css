.DraftItem {
  border-bottom: 1px solid;
  border-color: rgba(255,255,255,.12);
  cursor: pointer;
  display: flex;
}

.DraftItem:hover {
  background: #424242;
}

.DraftItem > div {
  box-sizing: border-box;
  padding: .75rem 1.5rem;
  padding-right: 0;
}

.DraftItem-selected {
  background-image: linear-gradient(rgba(255,255,255,.2), rgba(255,255,255,.2));  
}
