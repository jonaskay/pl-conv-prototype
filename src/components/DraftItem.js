import React from 'react';
import './DraftItem.css';

class DraftItem extends React.Component {
  render() {
    const item = this.props.item;
    
    return (
      <div className={'DraftItem' + (this.props.selected ? ' DraftItem-selected' : '')}>
        <div style={{flex: 'none', width: '13.5rem'}}>{item.name}</div>
        <div>{item.artist}</div>
      </div>
    );
  }
}

export default DraftItem;
