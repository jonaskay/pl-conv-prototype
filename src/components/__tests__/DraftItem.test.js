import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import DraftItem from '../DraftItem';

const item = { 
  "id": 1,
  "name": "Bodak Yellow", 
  "artist": "Cardi B",
  "image": "https://i.scdn.co/image/c6f9f7b2e71ae63388ec99bcc658eb173ed97bcc",
  "spotify_id": "2771LMNxwf62FTAdpJMQfM",
  "youtube_id": "PEGccV-NOm8"
}

it('renders a draft item', () => {
  const component = ReactTestRenderer.create(<DraftItem item={item} />);
  expect(component).toMatchSnapshot();
});

it('renders a selected draft item', () => {
  const component = ReactTestRenderer.create(<DraftItem item={item} selected={true} />);
  expect(component).toMatchSnapshot();
});
