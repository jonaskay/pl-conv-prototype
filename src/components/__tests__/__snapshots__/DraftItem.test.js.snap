// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a draft item 1`] = `
<div
  className="DraftItem"
>
  <div
    style={
      Object {
        "flex": "none",
        "width": "13.5rem",
      }
    }
  >
    Bodak Yellow
  </div>
  <div>
    Cardi B
  </div>
</div>
`;

exports[`renders a selected draft item 1`] = `
<div
  className="DraftItem DraftItem-selected"
>
  <div
    style={
      Object {
        "flex": "none",
        "width": "13.5rem",
      }
    }
  >
    Bodak Yellow
  </div>
  <div>
    Cardi B
  </div>
</div>
`;
